src/item/Elytra.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\item;
+
+use pocketmine\entity\Human;
+use pocketmine\entity\Living;
+
+class Elytra extends Armor{
+
+	private const MINIMUM_PITCH = -59;
+	private const MAXIMUM_PITCH = 38;
+
+	private const APPLY_DAMAGE_INTERVAL = 21;
+	private int $applyDamageTime = self::APPLY_DAMAGE_INTERVAL;
+
+	public function onTickWorn(Living $entity) : bool {
+		if ($entity instanceof Human && $entity->isGliding()) {
+
+			$this->applyDamageTime--;
+			if($this->applyDamageTime <= 0){
+				$this->applyDamage(1);
+				$this->applyDamageTime = self::APPLY_DAMAGE_INTERVAL;
+			}
+
+			$pitch = $entity->getLocation()->pitch;
+			// Check if the player's pitch is within the allowed range between -59 and 38.
+			if ($pitch >= self::MINIMUM_PITCH && $pitch <= self::MAXIMUM_PITCH) {
```

Comment:
This should be calculated instead of checking the pitch. You can probably find the calculation in the Java source code ot it's probably if the Y velocity is greater than the XZ velocity.

Reply:
> $velocity = $entity->getMotion();
> $yVelocity = $velocity->y;
> $xzVelocity = sqrt($velocity->x ** 2 + $velocity->z ** 2);
> 
> if ($yVelocity > $xzVelocity) {
> }

Like this? 🤔

Reply:
Based on tests fall damage is reset if the vertical velocity is less than 0.5 and those angles are what make it less than that. You can't rely on motion for the velocity because players always have 0 motion server-side.